migrations/sql/R__stored_procedure_for_permit_and_permitee_information.sql

```diff
@@ -576,77 +578,149 @@ CREATE OR REPLACE FUNCTION transfer_premit_permitee_information() RETURNS void A
         BEGIN
             RAISE NOTICE '.. Step 2 of 4: Update permit info';
             SELECT count(*) FROM permit INTO old_row;
+            SELECT count(*) FROM permit_amendment INTO old_amendment_row;
 
             -- Upsert permit data from ETL_PERMIT
-            RAISE NOTICE '.. Update existing records with latest MMS data';
+            RAISE NOTICE '.. Update existing permit records with latest MMS data';
             WITH updated_rows AS (
             UPDATE permit
             SET
-                received_date          = etl.received_date         ,
-                issue_date             = etl.issue_date            ,
                 update_user            = 'mms_migration'           ,
                 update_timestamp       = now()                     ,
-                authorization_end_date = etl.authorization_end_date,
                 permit_status_code     = etl.permit_status_code
             FROM ETL_PERMIT etl
             WHERE
                 permit.mine_guid = etl.mine_guid
                 AND
                 permit.permit_guid = etl.permit_guid
+				AND
+				issue_date = (select max(issue_date) from ETL_PERMIT where etl.permit_no = ETL_PERMIT.permit_no)
             RETURNING 1
             )
             SELECT COUNT(*) FROM updated_rows INTO update_row;
             RAISE NOTICE '....# of records in permit: %', old_row;
             RAISE NOTICE '....# of records updated in permit: %', update_row;
+			
+            -- Upsert permit amendment data from ETL_PERMIT
+            RAISE NOTICE '.. Update existing permit amendment records with latest MMS data';
+            WITH updated_rows AS (
+            UPDATE permit_amendment
+            SET
+                received_date          = etl.received_date         ,
+                issue_date             = etl.issue_date            ,
+                update_user            = 'mms_migration'           ,
+                update_timestamp       = now()                     ,
+                authorization_end_date = etl.authorization_end_date
+            FROM ETL_PERMIT etl
+            WHERE
+                permit_amendment.permit_amendment_guid = etl.permit_amendment_guid
+            RETURNING 1
+            )
+            SELECT COUNT(*) FROM updated_rows INTO update_amendment_row;
+            RAISE NOTICE '....# of records in permit: %', old_amendment_row;
+            RAISE NOTICE '....# of records updated in permit: %', update_amendment_row;
 
             RAISE NOTICE '.. Insert new MMS ETL_PERMIT records into permit';
             WITH
             --Select only new entry in ETL_PERMIT table
             new_permit AS (
                 SELECT *
-                FROM ETL_PERMIT
+                FROM ETL_PERMIT etl
                 WHERE permit_guid NOT IN (
                     SELECT permit_guid
                     FROM permit
                 )
+				AND issue_date = (select max(issue_date) from ETL_PERMIT where etl.permit_no = ETL_PERMIT.permit_no)
```

Comment:
Weird indentation?

Reply:
It doesn't show up like this locally in either VSCode or Notepad++ 👎